smarty/sampler.py

```diff
@@ -129,41 +144,16 @@ def __init__(self, molecules, basetypes_filename, initialtypes_filename, decorat
                 self.atomtypes = [(smarts, typename)] + self.atomtypes
```

Comment:
> Shouldn't these be listed as most general to least general? so the [*] option would come first.
That should fix this. Unless you want something to order the basetypes list. And if that is true:
> There are literately an infinite possibility of base types, I'm not sure what you're asking for.
classify them is not trivial.